src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -121,11 +123,17 @@ internal void TdsLogin(
             // length in bytes
             int length = TdsEnums.SQL2005_LOG_REC_FIXED_LEN;
 
-            string clientInterfaceName = TdsEnums.SQL_PROVIDER_NAME;
+            // Construct client interface name in format: Microsoft SqlClient - {OS}, {Platform} - {architecture}
+            // e.g. "Microsoft SqlClient - Microsoft Windows 10.0.26100, .NET 8.0.11 - X64"
+            string clientInterfaceName = new StringBuilder(TdsEnums.SQL_PROVIDER_NAME)
+                .Append(" - ").Append(GetOSInfo())
+                .Append(", ").Append(RuntimeInformation.FrameworkDescription ?? TdsEnums.UNKNOWN)
+                .Append(" - ").Append(RuntimeInformation.ProcessArchitecture)
+                .ToString();
+
             Debug.Assert(TdsEnums.MAXLEN_CLIENTINTERFACE >= clientInterfaceName.Length, "cchCltIntName can specify at most 128 unicode characters. See Tds spec");
```

Comment:
@cheenamalhotra  do you think its a good idea to have any additional safeguards to avoid even accidentally spilling over `MAXLEN_CLIENTINTERFACE` ?

Reply:
Perhaps, get rid of some information like Framework information if we spill over? 
The problem is that if a user runs into this problem, then they cannot recover from it, with any configuration changes. So a self correctly strategy would make sense. 

There is a lot of information here, we are getting from the runtime, which we dont control. So in that slight chance that we could spill over the Max len, a guardrail should be useful.